.github/workflows/build-test-e2e.yml

```diff
@@ -79,7 +81,8 @@ jobs:
         run: colima stop
   merge-reports:
     # Merge reports after playwright-tests, even if some shards have failed
-    if: always()
+    # But skip this job if the previous job was cancelled
+    if: ${{ !cancelled() }}
```

Comment:
If playwright-tests is skipped, then skip this job too. Otherwise, run this job regardless of whether playwright-tests succeeds or fails